### PR TITLE
DRAFT: Feature/icr-api-integration

### PR DESCRIPTION
**Pull request by psparacino**
Base working branch for ICR app and api PRs

APP PR: #2052 

API PR: #2051 

All changes in the PRs are reflected here

### CODE REVIEW

File: carbonmark/components/pages/Portfolio/Retire/utils/retire.ts [hunk 1]
```diff
@@ -65,11 +71,14 @@ export const retireProjectTokenTransaction = async (params: {
...
   tokenAddress: string;
+  tokenId: string;
+  tokenStandard: string;
   signer: providers.JsonRpcSigner;
   quantity: string;
   beneficiaryAddress: string;
   beneficiaryName: string;
   retirementMessage: string;
+  network: "polygon" | "mumbai";
   onStatus: OnStatusHandler;
```
Comment (psparacino) [inline carbonmark/components/pages/Portfolio/Retire/utils/retire.ts:82]: change to NetworkParam

File: carbonmark/components/CreateListing/index.tsx [hunk 2]
```diff
@@ -96,6 +100,9 @@ export const CreateListing: FC<Props> = (props) => {
    */
   const hasApproval = () => {
     if (!Number(inputValues?.amount)) return false;
+    if (!!inputValues?.tokenId) {
```
Comment (Atmosfearful) [inline carbonmark/components/CreateListing/index.tsx:103]: This needs a "why" comment
-- 1155's are "all or nothing" approvals across all tokenIds

File: carbonmark/lib/actions.ts [hunk 3]
```diff
@@ -145,7 +208,7 @@ export const createListingTransaction = async (params: {
       provider: signer,
       network: getSignerNetwork(signer) === "mumbai" ? "testnet" : "mainnet",
     });
-
+    console.log("createListing params", params);
```
Comment (Atmosfearful) [inline carbonmark/lib/actions.ts:211]: remove

**Review by Atmosfearful — APPROVED** [review]
This PR Was Pair Reviewed™️

NICE WORK 💯

**Pull request closed without merging**